functions/Set-DbaPrivilege.ps1

```diff
@@ -92,45 +92,80 @@ function Convert-UserNameToSID ([string] `$Acc ) {
+                                        $NewContent = @"
+[Privilege Rights]
+SeManageVolumePrivilege = *$SID
+"@
+                                        (Get-Content $tempfile) -replace "[Privilege Rights]", $NewContent | Set-Content $tempfile
+                                        Write-Verbose "Added new line for Instant File Initialization Privileges and granted $acc the privilege on $env:ComputerName" 
                                     }
                                 }
                             }
                             if ('LPIM' -in $Type) {
-                                $LPIMline = Get-Content $tempfile | Where-Object { $_ -match "SeLockMemoryPrivilege" }
                                 ForEach ($acc in $SQLServiceAccounts) {
                                     $SID = Convert-UserNameToSID -Acc $acc;
-                                    if ($LPIMline -notmatch $SID) {
-                                        (Get-Content $tempfile) -replace "SeLockMemoryPrivilege = ", "SeLockMemoryPrivilege = *$SID," |
-                                        Set-Content $tempfile
-                                        Write-Verbose "Added $acc to Lock Pages in Memory Privileges on $env:ComputerName"
-                                    }
+                                    $LPIMline = Get-Content $tempfile | Where-Object { $_ -match "SeLockMemoryPrivilege" }
+                                    if ($LPIMline) {
+                                        if ($LPIMline -notmatch $SID) {
+                                            (Get-Content $tempfile) -replace "SeLockMemoryPrivilege = ", "SeLockMemoryPrivilege = *$SID," |
+                                            Set-Content $tempfile
+                                            Write-Verbose "Added $acc to Lock Pages in Memory Privileges on $env:ComputerName"
+                                        }
+                                        else {
+                                            Write-Warning "$acc already has Lock Pages in Memory Privilege on $env:ComputerName"
+                                        }
+                                    }               
                                     else {
-                                        Write-Warning "$acc already has Lock Pages in Memory Privilege on $env:ComputerName"
-                                    }
+                                        #here-strings require the tabbing to get all messed up.
+                                        $NewContent = @"
+[Privilege Rights]
+SeLockMemoryPrivilege = *$SID
+"@
+                                        (Get-Content $tempfile) -replace "[Privilege Rights]", $NewContent | Set-Content $tempfile
+                                        Write-Verbose "Added new line for Lock Pages In Memory Privileges and granted $acc the privilege on $env:ComputerName" 
+                                    }       
                                 }
```

Comment:
Tested in my lab and this does not update the LPIM.

I played with this last night for a while as well and never could get it to work. The file won't ouput properly with the update.

Reply:
Let me check today if I get chance. The original work I did was outside this function and worked using the same method (and disgusting here-string!); I'm thinking I didn't remove the privilege before going back and testing this so it looked like it worked but hadn't done anything.

Reply:
thank you both so much 🍕🍔!